tests/test_labelarray.py

```diff
@@ -337,3 +339,68 @@ def test_setitem_array(self):
         # Write the whole array.
         arr[:] = orig_arr
         check_arrays(arr, orig_arr)
+
+    def test_narrow_code_storage(self):
+        def create_categories(width, plus_one):
+            length = int(width / 8) + plus_one
+            return [
+                ''.join(cs)
+                for cs in take(
+                    2 ** width + plus_one,
+                    product([chr(c) for c in range(256)], repeat=length),
+                )
+            ]
+
+        # uint8
+        categories = create_categories(8, plus_one=False)
+        arr = LabelArray(
+            [],
+            missing_value=categories[0],
+            categories=categories,
+        )
+        self.assertEqual(arr.itemsize, 1)
+
+        # uint8 inference
```

Comment:
For these cases, can we confirm that the label array with the inferred categories round-trips correctly back to strings?  Right now these tests are just asserting that the size is expected, but if we messed up our expected size calculations, then it's possible that we could overflow/underflow in the array, which would cause us to produce incorrect values.